src/chains/ethereum/ethereum/src/api.ts

```diff
@@ -550,19 +551,28 @@ export default class EthereumApi implements Api {
    */
   @assertArgLength(0, 1)
   async evm_setTime(time: number | QUANTITY | Date) {
-    let t: number;
+    let timestamp: number;
     switch (typeof time) {
       case "object":
-        t = time.getTime();
+        timestamp = time.getTime();
         break;
       case "number":
-        t = time;
+        timestamp = time;
         break;
       default:
-        t = Quantity.from(time).toNumber();
+        timestamp = Quantity.from(time).toNumber();
         break;
     }
-    return Math.floor(this.#blockchain.setTime(t) / 1000);
+    const blockchain = this.#blockchain;
+    // when using clock time use Date.now(), otherwise use the timestamp of the
+    // current latest block
+    const currentTime =
+      this.#options.miner.timestampIncrement === "clock"
```

Comment:
Would it make more sense for this logic to get the `sourceTime` to be inside the `setTimeDiff` function?
i.e.
```TypeScript
public setTimeDiff(newTime: number) {
  const sourceTime = this.#options.miner.timestampIncrement === "clock"
        ? Date.now()
        : this.blocks.latest.header.timestamp.toNumber() * 1000;
  return (this.#timeAdjustment = newTime - sourceTime);
}
```

Reply:
Good call! Done!